x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -246,6 +250,35 @@ public Object value(Aggregation agg, Map<String, String> fieldTypeMap, String lo
         }
     }
 
+    static class MultiBucketsAggExtractor implements AggValueExtractor {
```

Comment:
:+1: this is almost exactly the way I implemented it, too (well, there is probably no other way).

Reply:
Great minds think alike.